### ISSUE TITLE
add a `padding_idx` to Embedding

### ISSUE THREAD

**Issue by CarloLucibello** [FluxML/Flux.jl#2684]
Similar to [pytorch's behavior](https://docs.pytorch.org/docs/2.12/generated/torch.nn.Embedding.html#embedding), this would stop the gradient propagation to the embedding in position `padding_idx`, and would also initialize the embedding to all zeros.
